### ISSUE TITLE
Orchard 1.9 export fails

### ISSUE THREAD

**Issue by loshuliveros** [OrchardCMS/Orchard#5396]
I checked all options on Export page except botom 2 checkboxes - Roles and Workflows and after clicking on Export i got:

Value cannot be null. Parameter name: value
System.ArgumentNullException: Value cannot be null. Parameter name: value at System.Xml.Linq.XAttribute..ctor(XName name, Object value) at Orchard.Search.Drivers.SearchSettingsPartDriver.Exporting(SearchSettingsPart part, ExportContentContext context) at Orchard.ContentManagement.Drivers.ContentPartDriver`1.Orchard.ContentManagement.Drivers.IContentPartDriver.Exporting(ExportContentContext context) in d:\Projects\_moje\Orchard-1.9\src\Orchard\ContentManagement\Drivers\ContentPartDriver.cs:line 114 at Orchard.ContentManagement.Drivers.Coordinators.ContentPartDriverCoordinator.Exporting(ExportContentContext context) in d:\Projects\_moje\Orchard-1.9\src\Orchard\ContentManagement\Drivers\Coordinators\ContentPartDriverCoordinator.cs:line 83 at Orchard.ContentManagement.DefaultContentManager.Export(ContentItem contentItem) in d:\Projects\_moje\Orchard-1.9\src\Orchard\ContentManagement\DefaultContentManager.cs:line 794 at Orchard.ImportExport.Services.ImportExportService.ExportContentItem(ContentItem contentItem) at Orchard.ImportExport.Services.ImportExportService.ExportData(IEnumerable`1 contentTypes, IEnumerable`1 contentItems, Nullable`1 batchSize) at Orchard.ImportExport.Services.ImportExportService.Export(IEnumerable`1 contentTypes, IEnumerable`1 contentItems, ExportOptions exportOptions) at Orchard.ImportExport.Services.ImportExportService.Export(IEnumerable`1 contentTypes, ExportOptions exportOptions) at Orchard.ImportExport.Controllers.AdminController.ExportPOST() at lambda_method(Closure , ControllerBase , Object[] ) at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters) at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters) at System.Web.Mvc.Async.AsyncControllerActionInvoker.<BeginInvokeSynchronousActionMethod>b__39(IAsyncResult asyncResult, ActionInvocation innerInvokeState) at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`2.CallEndDelegate(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3d() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass46.<InvokeActionMethodFilterAsynchronouslyRecursive>b__3f()

I played with it and problem is only with "Site" checkbox from first section and "Site Settings" from second section:
"Site" in combination with "Data" checkbox.
"Site Settings" always.

**Comment by Piedone** [comment]
This is because the new Search settings in 1.9 weren't saved before. This is an issue if you upgrade from 1.9 or if you haven't specified settings after enabling the module.

**Comment by loshuliveros** [comment]
Yes, you're right, thank you. Setting indexing and then searching in site settings solved export problem.

**Comment by Piedone** [comment]
Maybe we should still do something with this so you don't have to face this issue after an upgrade.
